packages/lib/gpc/src/gpcUtil.ts

```diff
@@ -240,17 +339,21 @@ export type GPCRequirements = {
    */
   merkleMaxDepth: number;
 
+  /**
+   * Number of lists to be included in proof.
+   */
+
+  nLists: number;
```

Comment:
This boils down to allowing multiple entry values to be members of the same list, which means multiplicities have to be accounted for.

Reply:
But we're operating at the circuit level here, by which point the multiplicities have already been exploded?  Oh, I guess the Record below hides that since there can't be multiple entries for the same name.  Should it maybe be a `Record<PODName, [number, number]>` with each pair of numbers representing N copies of a list of length M?

Reply:
I had things set up that way initially, but as far as circuit selection is concerned all that matters is the largest list size and the total number of lists, which is why I figured I would leave the multiplicities out in favour of keeping track of the number of lists. Technically we don't even need the record and could just pass the max list size in.

Reply:
Maybe it's best to simplify back down to nLists and maxListSize then?  The checking that the names match up happens separately, and doesn't need to be related to circuit requirements.
Lists do seem somewhat different from tuples, where you're splitting them up so you do need to know individual arities.